internal/pioconfig/provenanceconfig.go

```diff
@@ -1,12 +1,63 @@
 package pioconfig
 
+import (
+	"fmt"
+)
+
 const (
-	// DefaultBondDenom is the denomination of coin to use for bond/staking
-	DefaultBondDenom = "nhash" // nano-hash
-	// DefaultFeeDenom is the denomination of coin to use for fees
-	DefaultFeeDenom = "nhash" // nano-hash
-	// DefaultMinGasPrices is the minimum gas prices coin value.
-	DefaultMinGasPrices = "1905" + DefaultFeeDenom
+	// defaultBondDenom is the denomination of coin to use for bond/staking
+	// should only be via provConfig variable
+	defaultBondDenom = "nhash" // nano-hash
+	// defaultFeeDenom is the (default) denomination of coin to use for fees
+	defaultFeeDenom = "nhash" // nano-hash
+	// defaultMinGasPrices is the (default) minimum gas prices integer value only
+	defaultMinGasPrices = 1905
 	// DefaultReDnmString is the allowed denom regex expression
 	DefaultReDnmString = `[a-zA-Z][a-zA-Z0-9/\-\.]{2,127}`
 )
+
+type ProvenanceConfig struct {
+	FeeDenom               string
+	ProvenanceMinGasPrices string // maps to defaultMinGasPrices in previous code,Node level config that provenance binary set's from appOpts.
+	// Current it will mirror MsgFeeFloorGasPrice.
+	MsgFeeFloorGasPrice int64 // Msg fee ante handlers and code use this for their calculations, this ***ONLY SETS***
+	// the default param(see method DefaultFloorGasPrice), all calculated values are still from msg fee module PARAMS.
+	// for that module, if the param is changed via governance then the code will pick the new value.(should pick that up from module param)
+	BondDenom     string // Also referred to as Staking Denom sometimes.
+	MsgFloorDenom string // MsgFloorDenom should always be the same Fee Denom, but maybe useful for tests.
+}
+
+var provConfig *ProvenanceConfig
+
+// SetProvenanceConfig in running the app it is called once from root.go. We decided not to seal it because we have tests,
+// which set the Config to test certain msg fee flows.
+// But the contract remains that this will be called once from root.go while starting up.
+func SetProvenanceConfig(customDenom string, msgFeeFloorGasPrice int64) {
+	if len(customDenom) > 0 && customDenom != defaultFeeDenom {
+		provConfig = &ProvenanceConfig{
+			FeeDenom:               customDenom,
+			ProvenanceMinGasPrices: fmt.Sprintf("%v", msgFeeFloorGasPrice) + customDenom,
+			MsgFeeFloorGasPrice:    msgFeeFloorGasPrice,
+			BondDenom:              customDenom,
+			MsgFloorDenom:          customDenom,
+		}
+	} else {
+		provConfig = &ProvenanceConfig{
+			FeeDenom: defaultFeeDenom,
+			// for backwards compatibility when these flags were not around, nhash will maintain behavior.
+			ProvenanceMinGasPrices: fmt.Sprintf("%v", defaultMinGasPrices) + defaultFeeDenom,
+			MsgFeeFloorGasPrice:    defaultMinGasPrices,
+			BondDenom:              defaultBondDenom,
+			MsgFloorDenom:          defaultFeeDenom,
+		}
+	}
```

Comment:
since this code replaces https://github.com/provenance-io/provenance/blob/cf5a1d3563499b9c089c599561084827f570cc91/x/msgfees/types/params.go#L13 which was already assuming  that param would be initially set as 1905nhash without exception for `nhash` (also just a param and can and probably will change soon hopefully in  non local networks) , but i like your change if we want to start a zone with nhash( which would practically an interesting decision..maybe) would be useful i guess... commited a version of ^^ code since above will not allow setting of 0vspn as floor price(one of the current requirements of usdf network) but will get set as 1905vspn, which is not the desired behavior in this case.